### PR TITLE
CIV-2525 HealthKey credential and claims support

### PR DESCRIPTION
**Pull request by jpsantosbh**
This adds the new HealthKey credentials and claims.
To support that a new Array Type support was implemented in this PR too.

### CODE REVIEW

File: src/claim/Claim.js [hunk 1]
```diff
@@ -187,18 +241,31 @@ class Claim extends UserCollectableAttribute {
   static getPath(identifier) {
     const { identifierComponents } = getBaseIdentifiers(identifier);
     const baseName = _.camelCase(identifierComponents[1]);
-    return `${baseName}.${identifierComponents[2]}`;
+    return baseName !== 'type' ? `${baseName}.${identifierComponents[2]}` : identifierComponents[2];
```
Comment (kevinhcolgan) [inline src/claim/Claim.js:244]: why do we act specifically on the name 'type' here?

File: __test__/creds/VerifiableCredential.test.js [hunk 2]
```diff
@@ -1362,7 +1465,7 @@ describe('Unit tests for Verifiable Credentials', () => {
         ));
         const ucaJson = SchemaGenerator.buildSampleJson(ucaDefinition);
         let value = ucaJson;
-        if (Object.keys(ucaJson).length === 1) {
+        if (Object.keys(ucaJson).length === 1 && ucaDefinition.type !== 'Array') {
```
Comment (dmelosantos) [inline __test__/creds/VerifiableCredential.test.js:1468]: 'Array' in a constant

File: __test__/creds/VerifiableCredentialSchema.test.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ describe('VerifiableCredentials SchemaGenerator validation', () => {
         ));
         const ucaJson = SchemaGenerator.buildSampleJson(ucaDefinition);
         let value = ucaJson;
-        if (Object.keys(ucaJson).length === 1) {
+        if (Object.keys(ucaJson).length === 1 && ucaDefinition.type !== 'Array') {
```
Comment (dmelosantos) [inline __test__/creds/VerifiableCredentialSchema.test.js:37]: 'Array' in a constant

File: src/schemas/generator/SchemaGenerator.js [hunk 4]
```diff
@@ -222,6 +226,8 @@ const generateRandomValueForType = (definition, includeDefinitions = false) => {
       return generateRandomNumberValueWithRange(refDefinition);
     case 'Boolean':
       return (Math.round(Math.random()) === 1);
+    // case 'Array':
```
Comment (kevinhcolgan) [inline src/schemas/generator/SchemaGenerator.js:229]: why is this commented out?

File: src/claim/Claim.js (hunk 1 above)
Comment (dmelosantos) [reply src/claim/Claim.js:244]: my question also

File: src/schemas/generator/SchemaGenerator.js [hunk 5]
```diff
@@ -222,6 +226,8 @@ const generateRandomValueForType = (definition, includeDefinitions = false) => {
...
+    //   return [makeJsonRecursion(refDefinition.items.type, includeDefinitions)];
```
Comment (dmelosantos) [inline src/schemas/generator/SchemaGenerator.js:230]: Is the schema generation not working?

**Review by dmelosantos — APPROVED** [review]

File: src/claim/Claim.js (hunk 1 above)
Comment (jpsantosbh) [reply src/claim/Claim.js:244]: Types need a special handle cause they are not include in the claim path ever

File: src/schemas/generator/SchemaGenerator.js (hunk 5 above)
Comment (jpsantosbh) [reply src/schemas/generator/SchemaGenerator.js:230]: It works without that

**Pull request merged**